atd-vze/src/Components/GridTable.js

```diff
@@ -98,6 +121,69 @@ const GridTable = ({ title, query }) => {
    */
   const clearFilters = () => {
     setSearchParameters({});
+    setFilterOptions({});
+  };
+
+  /**
+   * Returns true if the input string is a valid alphanumeric object key
+   * @param {string} input - The string to be tested
+   * @returns {boolean}
+   */
+  const isAlphanumeric = input =>
+    input.match(/^[0-9a-zA-Z\-\_]+$/) === null ? false : true;
+
+  /**
+   * Extracts a list of keys in a graphql expression
+   * @param {string} exp - The expression
+   * @returns {Array}
+   */
+  const listKeys = exp =>
+    exp.split(/[\{\} ]+/).filter(n => isAlphanumeric(n) && n !== "");
+
+  /**
+   * Returns the value of a data structure based on the list of keys provided
+   * @param {object} obj - the object in question
+   * @param {Array} keys - the list of keys
+   * @returns {*}
+   */
+  const resVal = (obj, keys) => {
```

Comment:
As my old mentor used to tell me, "We're not a start up, we can afford to spell out our function names" 

The idea being, you don't know what `resVal` is until you read the code or your comment. But `responseValue` makes a little more sense.

Reply:
Refactored to responseValue